### PR TITLE
Allow to override timeout in case of slow network

### PR DESCRIPTION
**Pull request by AlbanAndrieu**
Allow to override timeout in case of slow network

System.setProperty("http.socket.timeout", "600")

### Your checklist for this pull request

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side) and not your master branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or in [Jenkins JIRA](https://issues.jenkins-ci.org)
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Did you provide a test-case? That demonstrates feature works or fixes the issue.

Tested with Jenkins ver. 2.204.3 and Atlassian Bitbucket v4.5.2 on Ubuntu 18 and MacOSX

---
This change is [Reviewable](https://reviewable.io/reviews/jenkinsci/bitbucket-branch-source-plugin/292)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -174,6 +176,26 @@
       <artifactId>authentication-tokens</artifactId>
       <version>1.3</version>
     </dependency>
+    <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>log4j-over-slf4j</artifactId>
+      <version>${slf4j.version}</version>
```
Comment (Jetersen) [inline pom.xml]: These deps should be coming from Jenkins.

File: pom.xml [hunk 2]
```diff
@@ -174,6 +176,26 @@
...
+    </dependency>
+    <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-jdk14</artifactId>
```
Comment (Jetersen) [inline pom.xml]: 😱 Jenkins only builds with JDK 8 and JRE 11 runtime is supported.

**Comment by AlbanAndrieu** [comment]
Where shall the documentation? Shall we add that we can now override socket timeout?System.setProperty("http.socket.timeout", "600")

**Comment by Jetersen** [comment]
There is the docs folder here on github feel free to add it there.

File: pom.xml [hunk 3]
```diff
@@ -51,9 +51,11 @@
   <properties>
     <revision>2.7.1</revision>
     <changelist>-SNAPSHOT</changelist>
+    <!--<jenkins.version>2.221</jenkins.version>-->
     <jenkins.version>2.138.4</jenkins.version>
     <java.level>8</java.level>
     <hpi.compatibleSinceVersion>2.0</hpi.compatibleSinceVersion>
+    <slf4j.version>1.7.26</slf4j.version>
```
[the remark this answers is not in the document]
Comment (AlbanAndrieu) [reply pom.xml]: slf4j removed

**Comment by AlbanAndrieu** [comment]
Windows build seems unstable

**Comment by AlbanAndrieu** [comment]
Hi @Jetersen 

Shall I need to do anythings else to merge and close this pull request?

Feel free to tell me.

**Comment by Jetersen** [comment]
Waiting for Jenkins CI to pass the build. Best way to trigger the CI is by closing and opening the PR.

**Comment by jetersen** [comment]
This look like a failed merge in the changes.

**Comment by AlbanAndrieu** [comment]
> This look like a failed merge in the changes.

Indeed `git reset --hard HEAD~1` then `git push -f` seems to fix it.

**Comment by AlbanAndrieu** [comment]
@jetersen Do you need something else?

**Comment by jetersen** [comment]
No, I just need set a window to get some PRs merged, reviewed and released.

**Pull request merged**